### ISSUE TITLE
Mailchimp uses incorrect image url, when adding product block to email

### ISSUE THREAD

**Issue by ktsixit** [mailchimp/mc-magento2#1542]
### Preconditions

    1. Magento ver. 2.4.2
    2. ebizmarts/mailchimp-lib 3.0.34
    3. mailchimp/mc-magento2 103.4.44
    4. Integration magento-mailchimp is complete and working. Ecommerce is enabled

### Steps to reproduce

    1. Create a new campaign in mailchimp
    2. Insert a Product block
    3. Select product by name
    4. Product is added to template but has incorrect url.

Example of incorrect image url:
https://www.example.com/media/catalog/product/cache/1/image/265x/9df78eab33525d08d6e5fb8d27136e95/0/7/0719.057_main.jpg

### Actual and Expected result

Image path should be something like this:
https://www.example.com/media/catalog/product/cache/7a76cf03197d40fe0de363f36b889dd6/0/7/0719.057_main.jpg

### Additional information

There is no directory named "1" in "cache".
Path is completely incorrect and I have no idea where there urls come from or how magento extension submits it to mailchimp.

**Comment by gonzaloebiz** [comment]
Hi @ktsixit 

we never take the caches images, why? Because if you flush catalog images cache, the image will be obsolete and should need to re-sync all the product data.
So if in the frontend the image looks like http://mc-magento24.ebizmarts.com/media/catalog/product/cache/10073d9bf5e409240d6b91d79ffda06b/m/b/mb02-gray-0.jpg we use http://mc-magento24.ebizmarts.com/media/catalog/product/m/b/mb02-gray-0.jpg (points directly to the real image file and not to the cached one)
This is the default way that the extension works.
You can find here https://github.com/mailchimp/mc-magento2/blob/develop-2.3/Model/Api/Product.php#L398-L410 how we fill the product image, try to put some logs on there to know why your installation takes the cached images

Best

**Comment by ktsixit** [comment]
Thank you for that, Do I have to resync all products in order to debug code?

**Comment by ktsixit** [comment]
Using logs I managed to test code and it looks like, at this point, it generates non-cache image urls, like this:
`$data["image_url"] = https://www.example.com/media/catalog/product/i/m/image_PB036_1.jpg`

There's something afterwards that changes urls. Do you have any idea what that might be?

**Comment by gonzaloebiz** [comment]
Try to modify a product and wait until the product was re-synced, then try to make a campaign with this product and check the URL

Best

**Comment by ktsixit** [comment]
> Try to modify a product and wait until the product was re-synced, then try to make a campaign with this product and check the URL
> 
> Best

I tried that, but didn't work

**Comment by ktsixit** [comment]
Is there any chance that mailchimp would generate cache urls, by older Authorized applications?
**These image urls look like MAGENTO-1 urls.**

I used to have magento1 integration enabled (when website was on magento1) but past several months it has been moved to magento2.

![Connected-Sites-Mailchimp](https://user-images.githubusercontent.com/2758872/187442782-774a4d94-78eb-4cfb-a1c8-7138b5e000a9.png)

I have tried deleting mailchimp store and create NEW STORE using NEW API key. Everything looks ok in magento but in mailchimp I'm still getting wrong urls.
![Configuration-Settings-Stores-Magento-Admin (4)](https://user-images.githubusercontent.com/2758872/187443050-74fb9493-c845-48d9-b6da-5cd93bd795f0.png)
